internal/impl/nats/auth.go

```diff
@@ -248,3 +272,23 @@ func loadFileContents(filename string, fs *service.FS) ([]byte, error) {
 
 	return io.ReadAll(f)
 }
+
+func saveTempNkeyFile(content string) (string, error) {
```

Comment:
I think writing this file to disk is acceptible but we need to have it documented in the field description so that users are aware. Maybe add something like "Specify an NKey seed directly in config, this seed will be written to (and loaded from) a temporary file." to the description.

The other issue is that a new temporary file would be written upon each instantiation, which could potentially result in unbounded numbers of files being created during crashloops, batch jobs, dynamically run pipelines, and so on. We'd need to solve that either by having a consistent temporary file path, or just require that `nkey_file` is specified but the file is created when it doesn't already exist and this new field is populated, that allows users to choose where the file is written.

Reply:
Instead of creating a temp file, maybe try this instead:
```go
kp, err := nkeys.ParseDecoratedNKey(auth.NKey)
if err != nil {
    return nil, fmt.Errorf("failed to parse nkey")
}

pub, err := kp.PublicKey()
if err != nil {
    return nil, fmt.Errorf("failed to extract public key from nkey")
}
if !nkeys.IsValidPublicUserKey(pub) {
    return nil, fmt.Errorf("invalid nkey user seed")
}
sigCB := func(nonce []byte) ([]byte, error) {
	sig, _ := kp.Sign(nonce)
	return sig, nil
}

opts = append(opts, nats.Nkey(string(pub), sigCB))
```

I took a peek at their code: https://github.com/nats-io/nats.go/blob/278f9f188bca4d7bdee283a0e98ab66b82530c60/nats.go#L5782

Reply:
Thanks @mihaitodor ! I will give it a try.

Again, I'm just surprised that they don't have a `nats.NkeyOptionFromString` 😄 I thought this might be missing for a good reason.

Reply:
You’re welcome! I think they’d be open to adding it if you open an issue on their repo.

Reply:
Hi @mihaitodor , I've updated the code based on your comment so that it no longer stores anything in temp file.

I did setup local nats server and tested it, all work well.

Thanks!